hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -578,19 +703,43 @@ static int32_t executeCommand(struct mtd_k64f_data *context)
         /* Execution may result in failure. Check for errors */
         if (failedWithAccessError() || failedWithProtectionError()) {
             clearErrorStatusBits();
+            #if EXISTS_POSSIBILITY_OF_CONCURRENT_READ
+                __enable_irq();
+            #endif
             return ARM_DRIVER_ERROR_PARAMETER;
         }
         if (failedWithRunTimeError()) {
+            #if EXISTS_POSSIBILITY_OF_CONCURRENT_READ
+                __enable_irq();
+            #endif
             return ARM_DRIVER_ERROR; /* unspecified runtime error. */
         }
 
         /* signal synchronous completion. */
         switch (context->currentCommand) {
             case ARM_STORAGE_OPERATION_PROGRAM_DATA:
                 if (context->amountLeftToOperate == 0) {
+                    #if EXISTS_POSSIBILITY_OF_CONCURRENT_READ
+                        __enable_irq();
+                    #endif
                     return context->sizeofCurrentOperation;
                 }
 
+                /* Allow pending interrupts to be handled before launching the successive
+                 * operation. This avoids having to block interrupts for long periods.
+                 *
+                 * ProgramData operates in steps of 1KB; initial experiments
+                 * indicate that the time needed to complete a 1KB write is
+                 * around 6.5ms. As a programData progresses, interrupts would
+                 * need to be held blocked for periods of such duration
+                 * if EXISTS_POSSIBILITY_OF_CONCURRENT_READ.
+                 */
+                #if EXISTS_POSSIBILITY_OF_CONCURRENT_READ
+                    __enable_irq();
+                    /* service any pending interrupts at this moment */
```

Comment:
This in usually not sufficient to trigger interrupts. ARMs usually need interrupts enabled for a little while to let them trigger.

Is there an architecturally guaranteed procedure for this? I have a feeling there might be in ARMv7A+M, but can't find a reference. Maybe an ISB instruction or something.